### DISCUSSION TITLE
Help compare Comment and Annotation services: moderation, spam, notifications, configurability

### DISCUSSION THREAD

**Discussion by westurner** [executablebooks/meta#102] [Q&A]
jupyter-comment supports a number of commenting services (https://github.com/executablebooks/sphinx-comments/issues/14 , https://github.com/executablebooks/jupyter-book/issues/861). In helping users decide which commenting and annotation services to include on their pages and commit to maintaining, could we discuss criteria for assessment and current features of services?

Possible features for comparison:

- Content author can delete / hide
- Content author can report / block
- Comments / annotations are screened by spam-fighting service
- Content / author can label as e.g. toxic
- Content author receives notification of new comments
- Content author can require approval before user-contributed content is publicly-visible
- Content author may allow comments for a limited amount of time (probably more relevant to BlogPostings)
- Content author may simultaneously denounce censorship in all it's forms while allowing previously-published works to languish

**Comment by choldgraf** [comment]
I think this is a good idea - would love input from anybody in the community per their experience with these tools (I am not super familiar with them myself). Perhaps this is also something people can learn more about and share as people use these tools in their books!

**Comment by westurner** [comment]
> Open-source self-hosted comments
> Comparison table for open-source self-hosted commenting servers
https://lisakov.com/projects/open-source-comments/

The data for the site is stored in YAML:
https://github.com/pozitron57/open-source-comments/blob/master/data.yaml 

And the feature schema is here: https://github.com/pozitron57/open-source-comments/blob/master/yaml_2_js.py#L58 ::

```python
fields_dic={
    'stars':                 'Github stars',
    'stars_dif':             'Stars&nbsp;in 2&nbsp;weeks',
    'name':                  'Name',
    'source':                'Source code',
    'demo':                  'Demo & examples',
    'js_kB':                 'js, kB',
    'css_kB':                'css, kB',
    'language':              'Language',
    'db':                    'Database',
    'mail_notification':     'Mail notification',
    'edit':                  'User can edit',
    'vote':                  'User can vote',
    'antispam':              'Antispam',
    'bad_words_list':        'Bad words list',
    'use_cookies':           'Uses cookies',
    'avatar':                'Avatar',
    'markdown_support':      'Markdown support',
    'moderation':            'Moderation',
    'dependency':            'Dependencies',
    'webmention':            'Supports Webmention',
    'nested_comments':       'Nested comments',
    'provided_hosting':      'Can host for you',
    'collapse_comments':     'Collapse comments',
    'sort':                  'Configurable order',
    'rate_limit':            'Comment rate limit',
    'docker':                'Docker container',
    'paging':                'Paging',
    'hide_long_threads':     'Hide long threads',
    'anonymous_comments':    'Anonymous comments',
    'social_network_login':  'Social network login',
    'import_from_wordpress': 'Import from wordpress',
    'import_from_disqus':    'Import from disqus',
    'english_documentation': 'English documentation',
    'rss':                   'RSS',
    'display_images':        'Display images',
    'static':                'Static',
    'description':           'Description',
    'updated':               'Updated (y‑m‑d)',
    'last_committed':        'Updated (y‑m‑d)',
    'created':               'Created (y‑m‑d)',
    'license':               'License',
    'open_issues':           'Open issues + PR',
    'unmaintained':          'Unmaintained',
}
```

Found this on HN today.

**Comment by westurner** [comment]
https://jupyterbook.org/en/stable/interactive/comments.html#commenting-and-annotating
- Utterances
- Hypothesis
- https://sphinx-comments.readthedocs.io/en/latest/
  - Uterrances
  - Hypothesis
  - Dokie.li
